src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -1,127 +1,171 @@
 import json
 import os
 import shutil
+import time
 import uuid
 from pathlib import Path
-from warnings import WarningMessage
+from typing import List
 
 import datajoint as dj
+import numpy as np
 import spikeinterface as si
 import spikeinterface.toolkit as st
 
 from ..common.common_interval import IntervalList
 from ..common.common_nwbfile import AnalysisNwbfile
 from ..common.dj_helper_fn import fetch_nwb
-from .spikesorting_recording import SpikeSortingRecording
+from .merged_sorting_extractor import MergedSortingExtractor
+from .spikesorting_recording import SortInterval, SpikeSortingRecording
 from .spikesorting_sorting import SpikeSorting
 
 schema = dj.schema('spikesorting_curation')
 
+valid_labels = ['reject', 'noise', 'artifact', 'mua', 'accept']
+
+
+def apply_merge_groups_to_sorting(sorting: si.BaseSorting, merge_groups: List[List[int]]):
+    # return a new sorting where the units are merged according to merge_groups
+    # merge_groups is a list of lists of unit_ids.
+    # for example: merge_groups = [[1, 2], [5, 8, 4]]]
+
+    return MergedSortingExtractor(parent_sorting=sorting, merge_groups=merge_groups)
+
 
 @schema
 class Curation(dj.Manual):
     definition = """
     # Stores each spike sorting; similar to IntervalList
-    curation_id: varchar(10)
+    curation_id: int # a number correponding to the index of this curation
     -> SpikeSorting
     ---
-    parent_curation_id='': varchar(10)
-    labels: blob # a dictionary of labels for the units
+    parent_curation_id=-1: int
+    curation_labels: blob # a dictionary of labels for the units
     merge_groups: blob # a list of merge groups for the units
-    metrics: blob # a list of quality metrics for the units (if available)
+    quality_metrics: blob # a list of quality metrics for the units (if available)
```

Comment:
Seems like quality metrics are saved as json when `make` is called on `QualityMeric` table. Is this then redundant? or should `QualityMetric` not be dumping qm to json?